### DISCUSSION TITLE
[AKS] Dostęp do prywatnego Azure K8s przy użyciu Virtual Network Gateway

### DISCUSSION THREAD

**Discussion by norbertgruszka** [AzureCommunityPL/group#348] [Pomoc] [answered]
Cześć, 

potrzebował bym pomocy przy stawianiu Kubernetes na Azure. 

Postawiłem Private Kubernetes service i chciałem mieć do niego dostęp z lokalnej maszyny. Dostawiłem w tym samym VNET Virtual Network Gateway i skonfigurowałem połączenie point-to-site. Jestem w stanie ping-ować urządzenia w sieci, jednak nie mogę dobić się do k8s. Dla przykładu `kubectl get pods` zwraca:

```
Unable to connect to the server: dial tcp: lookup <<api-server-address>>: no such host
```

Dla sprawdzenia, dostawiłem jeszcze vm + bastion, wszystko w tym samym vnet. Tym razem już mogę się dobić do k8s i np wyświetlić pod-y, zainstalować helm charm itd. 

Podsumowując: 
- przy połączeniu przez Virtual Network Gateway kubectl nie rozpoznaje host-a 
- przy połączeniu przez bastion, kubectl rozpoznaje hosta.

Czy ktoś może wie dlaczego tak się dzieje? Byłby ktoś wstanie nakierować mnie jak to poprawnie skonfigurować?

Dzięki!

**Comment by ptrstpp950** [comment, marked as answer]
Problemem jest „brak” DNS wpiętego u Ciebie. Kilka poprawnych obejść jest opisanych na stronie: https://docs.microsoft.com/en-us/answers/questions/39571/not-resolving-private-dns-zone-over-point-to-site.html

Acz nie pamietam z głowy które pomaga. Możesz też oczywiście spróbować hack z wpisaniem sobie adresu w hosts nim zaczniesz żeby upewnić się że działa 😅

**Reply by norbertgruszka** [reply]
Udało się, faktycznie brakowało DNS po mojej stronie, ale musiałem jeszcze dostawić DNS Forwarder https://azure.microsoft.com/en-us/resources/templates/dns-forwarder/ . Mając to, mogłem podać adres VM-ki w konfiguracji XML dla VPN client:
```
<clientconfig>
  <dnsservers>
    <dnsserver>x.x.x.x</dnsserver>
  </dnsservers>
</clientconfig>
```

Dzięki za pomoc!
